src/Blocks/BlockIce.h

```diff
@@ -54,6 +54,7 @@ class cBlockIceHandler :
 		auto blockTypeBelow = a_ChunkInterface.GetBlock(a_BlockPos.addedY(-1));
 		if (cBlockInfo::FullyOccupiesVoxel(blockTypeBelow) || IsBlockLiquid(blockTypeBelow))
 		{
+			a_ChunkInterface.FastSetBlock(a_BlockPos, E_BLOCK_AIR, 0);
```

Comment:
This definitely warrants an explanatory comment - why `FastSetBlock` followed by a `SetBlock`. When casually browsing the code, I'd remove a construct like this (maybe I even did, in the past).